test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ModulesWithGlobalFallbackTest.scala

```diff
@@ -0,0 +1,205 @@
+  }
+
+  @Test def testImportClassInModule(): Unit = {
+    val b = new Buffer(5)
+    for (i <- 0 until 5)
+      b(i) = (i * i).toShort
+
+    for (i <- 0 until 5)
+      assertEquals(i * i, b(i).toInt)
+  }
+
+  @Test def testImportIntegrated(): Unit = {
+    val b = new Buffer(js.Array[Short](0xe3, 0x81, 0x93, 0xe3, 0x82, 0x93, 0xe3,
+        0x81, 0xab, 0xe3, 0x81, 0xa1, 0xe3, 0x81, 0xaf))
+    val decoder = new StringDecoder()
+    assertTrue(Buffer.isBuffer(b))
+    assertFalse(Buffer.isBuffer(decoder))
+    assertEquals("こんにちは", decoder.write(b))
+    assertEquals("", decoder.end())
+  }
+
+}
+
+object ModulesWithGlobalFallbackTest {
+  @ScalaJSDefined
+  private object QueryStringFallbackImpl extends js.Object {
+    def stringify(obj: js.Dictionary[String], sep: String = "&",
+        eq: String = "="): String = {
+      var result = ""
+      for ((key, value) <- obj) {
+        if (result != "")
+          result += sep
+        result += key + eq + value
+      }
+      result
+    }
+  }
+
+  @ScalaJSDefined
+  private class StringDecoderFallbackImpl(charsetName: String = "utf8")
+      extends js.Object {
+    import java.nio.charset._
+
+    private val charset = Charset.forName(charsetName)
+    private val decoder = charset.newDecoder()
+
+    private def writeInternal(buffer: Uint8Array,
+        endOfInput: Boolean): String = {
+      val in = TypedArrayBuffer.wrap(buffer.buffer, buffer.byteOffset,
+          buffer.byteLength)
+
+      // +2 so that a pending incomplete character has some space
+      val out = CharBuffer.allocate(
+          Math.ceil(decoder.maxCharsPerByte().toDouble * in.remaining()).toInt + 2)
+
+      val result = decoder.decode(in, out, endOfInput)
+      if (!result.isUnderflow())
+        result.throwException()
+
+      if (endOfInput) {
+        val flushResult = decoder.flush(out)
+        if (!flushResult.isUnderflow())
+          flushResult.throwException()
+      }
+
+      out.flip()
+      out.toString()
+    }
+
+    def write(buffer: Uint8Array): String =
+      writeInternal(buffer, endOfInput = false)
+
+    def end(buffer: Uint8Array): String =
+      writeInternal(buffer, endOfInput = true)
+
+    def end(): String =
+      writeInternal(new Uint8Array(0), endOfInput = true)
+  }
+
+  @ScalaJSDefined
+  object BufferStaticFallbackImpl extends js.Object {
+    def isBuffer(x: Any): Boolean = x.isInstanceOf[Uint8Array]
+  }
+
+  @BeforeClass
+  def beforeClass(): Unit = {
+    assumeTrue("Assuming that Typed Arrays are supported",
+        areTypedArraysSupported)
+
+    js.Dynamic.global.ModulesWithGlobalFallbackTest_QueryString =
+      QueryStringFallbackImpl
+    js.Dynamic.global.ModulesWithGlobalFallbackTest_StringDecoder =
+      js.constructorOf[StringDecoderFallbackImpl]
+    js.Dynamic.global.ModulesWithGlobalFallbackTest_Buffer =
+      js.constructorOf[Uint8Array]
+    js.Dynamic.global.ModulesWithGlobalFallbackTest_BufferStatic =
+      BufferStaticFallbackImpl
```

Comment:
I assume you have tested this, but I can't see why (except for `ModulesWithGlobalFallbackTest_Buffer`, but that one is also not defined in this file).